feat: Add CMEK encryption plugin

XuanYang-cn opened the pull request:
See also: #40321

aoiasd commented on internal/util/hookutil/default.go:
```diff
@@ -63,3 +63,35 @@ func (d DefaultExtension) Report(info any) int {
+	return plainText, nil
+}
+
+func (d DefaultCipher) Decrypt(cipherText []byte) (plainText []byte, err error) {
```
Encrypt and Decrypt was useless. Could remove them.

XuanYang-cn replied:
This is to impl Encryptor and Decryptor and Cipher in one DefaultCipher.

XuanYang-cn commented:
/run-cpu-e2e

tedxu commented on internal/util/hookutil/default.go:
```diff
@@ -63,3 +63,35 @@ func (d DefaultExtension) Report(info any) int {
 	return nil
 }
+
+type DefaultCipher struct{}
```
What is this object used for?

tedxu commented on internal/util/hookutil/default.go:
```diff
@@ -63,3 +63,35 @@ func (d DefaultExtension) Report(info any) int {
+	return nil
+}
+
+func (d DefaultCipher) GetEncryptor(ezID int64) (encryptor hook.Encryptor, safeKey []byte, err error) {
```
Do we need an additional level under EZ? I mean, if the `collection` level is invisible, we can't revoke data access by collection.

XuanYang-cn committed:
feature: Add CMEK cipher plugin

1. Enable Milvus to read cipher configs
2. Enable cipher plugin in binlog reader and writer
3. Add a testCipher for unittests

See also: #40321

Signed-off-by: yangxuan <xuan.yang@zilliz.com>

XuanYang-cn committed:
fix

Signed-off-by: yangxuan <xuan.yang@zilliz.com>
